app/models/project_group.rb

```diff
@@ -51,30 +54,46 @@ def summary_or_none
     summary.blank? ? "[#{I18n.t("none")}]" : summary.to_s
   end
 
-  # Gets the total number of steps beneath this group.
+  # Copies filters down through all descendant groups and resets memoization of filtered_children.
+  def filters=(filters)
+    @filters = filters
+    @filtered_children = nil # Undo memoization
+    children.each { |child| child.filters = filters if child.is_a?(ProjectGroup) }
+  end
+
+  def filtered_children
+    @filtered_children ||= children.reject do |child|
+      filters.present? && child.is_a?(ProjectStep) &&
```

Comment:
Maybe it would be better to do `child.is_project_step?` vs `child.is_a?(ProjectStep)`